src/elements/Header/Header.js

```diff
@@ -47,22 +47,29 @@ function Header(props) {
 
   if (icon && typeof icon !== 'boolean') {
     return (
-      <ElementType className={classes} {...rest}>
+      <ElementType {...rest} className={classes}>
         {createIcon(icon)}
         {content && <HeaderContent>{content}</HeaderContent>}
+        {subheader && <HeaderSubheader content={subheader} />}
       </ElementType>
     )
   }
 
   if (image) {
     return (
-      <ElementType className={classes} {...rest}>
+      <ElementType {...rest} className={classes}>
         {createImage(image)} {content}
+        {subheader && <HeaderSubheader content={subheader} />}
       </ElementType>
     )
   }
 
-  return <ElementType {...rest} className={classes}>{children || content}</ElementType>
+  return (
+    <ElementType {...rest} className={classes}>
+      {children || content}
+      {subheader && <HeaderSubheader content={subheader} />}
+    </ElementType>
+  )
```

Comment:
Currently, the Header will render children with shorthand even though there are prop type warnings.  While we're here, let's split this return with a conditional return to only render children if provided.  The goal in doing this in all components is to enforce proper usage (not rendering shorthand when children are present):

```jsx
  if (children) {
    return (
      <ElementType {...rest} className={classes}>
        {children}
      </ElementType>
    )
  }

  return (
    <ElementType {...rest} className={classes}>
      {content}
      {subheader && <HeaderSubheader content={subheader} />}
    </ElementType>
  )
```